File: Arrays 101/4. Duplicate Zeros/duplicate_zeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        temp = []
        for item in arr:
            temp.append(item)

        i = 0
        dst_idx = 0
        while i < len(temp):
            arr[i] = temp[dst_idx]
            i += 1
            if i == len(temp):
                break
            if temp[dst_idx] == 0:
                arr[i] = 0
                i += 1
            dst_idx += 1

File: Arrays 101/1089. Duplicate Zeros/duplicate_zeros.cpp
class Solution
{
public:
    void duplicateZeros(vector<int> &arr)
    {
        vector<int> temp;
        for (int i = 0; i < arr.size(); i++)
        {
            temp.push_back(arr[i]);
        }

        int dst_idx = 0;
        int i = 0;
        while (i < arr.size())
        {
            arr[i++] = temp[dst_idx];
            if (i == arr.size())
            {
                break;
            }
            if (temp[dst_idx] == 0)
            {
                arr[i++] = 0;
            }
            dst_idx++;
        }
    }
};